src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,176 @@
+            $pbjx->sendAt(
+                $retryCommand,
+                strtotime($this->flags->getString('inspect_seo_retry_delay', '+15 minutes')),
+                "$nodeRef.inspect-seo"
+            );
+        }
+    }
+
+    public function checkIndexStatusForGoogle(Message $command, Pbjx $pbjx, Message $node): Message
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_google_disabled')) {
+            return $command;
+        }
+
+        $request = new \Google_Service_SearchConsole_InspectUrlIndexRequest();
+        $siteUrl = $this->flags->getString('inspect_seo_google_site_url');
+
+        if (empty($siteUrl)) {
+            $this->logger->error('Must provide a site url.', [
+                'site_url' => $siteUrl,
+            ]);
+
+            return $command;
+        }
+
+        $request->setSiteUrl($siteUrl);
+        $url = UriTemplateService::expand(
+            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
+        );
+        $request->setInspectionUrl($url);
+        $client = new \Google_Client();
+        $client->addScope(\Google_Service_SearchConsole::WEBMASTERS_READONLY);
+
+        $response = null;
+
+        try {
+            $base64EncodedAuthConfig = Crypto::decrypt(getenv('GOOGLE_SEARCH_CONSOLE_API_SERVICE_ACCOUNT_OAUTH_CONFIG'), $this->key);
+            $client->setAuthConfig(json_decode(base64_decode($base64EncodedAuthConfig), true));
+            $service = new \Google_Service_SearchConsole($client);
+            $response = $service->urlInspection_index->inspect($request);
+        } catch (\Throwable $e) {
+            $this->logger->error('An error occurred in checkIndexStatus for google.', [
+                'exception' => $e,
+                'node_ref' => $node->generateNodeRef(),
+                'url' => $url,
+                'retry_count' => $command->get('ctx_retries'),
+                'stack_trace' => $e->getTraceAsString(),
+            ]);
+        }
+
+        if ($response === null) {
+            return $command;
+        }
+
+        $inspectionResult = $response->getInspectionResult();
+        $indexStatusResult = $inspectionResult->getIndexStatusResult() ?? null;
+        $ampResult = $inspectionResult->getAmpResult() ?? null;
+
+        $isConclusiveForWeb = $indexStatusResult && ($indexStatusResult->getVerdict() === 'PASS' || $indexStatusResult->getVerdict() === 'FAIL');
+        $isConclusiveForAmp = !$node::schema()->hasField('amp_enabled') || ($ampResult && ($ampResult->getVerdict() === 'PASS' || $ampResult->getVerdict() === 'FAIL'));
+
+        if ($isConclusiveForWeb && $isConclusiveForAmp) {
+            $this->putEvent($command, $pbjx, $node, 'google', $response);
+            return $command;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
+            $command->addToSet('search_engines', ['google']);
```

Comment:
@efjacobson @digitalartcrew i would expect a retry in this scenario is just for google, not for all search_engines.  so you'd want to clear the field first.  might need to run through this more in depth to ensure retry process is not redoing all the things.  also, a retry command with a job id (".inspect-seo" suffix) will cancel each others retries on multiple search engines.

Reply:
This looks correct. The field is clear before entering the command: https://github.com/triniti/core-php/pull/34/files/ca551b772e779dc043d7ac047f352a61365676a1#diff-6ab784c5caf2e67b8d9e4a618627f1ab2974c5e93b92d3797108a028448946faR58

A search engine is then added within each `checkIndexStatusFor{searchEngine}` if retries are not exceeded. With that said it does look like an issue where the retries are being summed for all search Engines versus per search engine

Reply:
@gdbrown @digitalartcrew i dont think there is an issue - each checkIndexForSearchEngine method will do one of:
- succeed: publish its own event, then return the unmodified command
- retry: return the command after having added itself (back) into the (previously cleared) set of search_engines
- fail: same as succeed, really, now that i look at it (it used to log something as well but i dont know that its necessary)